tools/local_fix_format.sh

```diff
@@ -44,13 +44,13 @@ fi
 
 # Runs the formatting functions on the specified args, echoing commands
 # if -vergbose was supplied to the script.
-function format_one() {
+function format_some() {
   (
     if [[ "$verbose" == "1" ]]; then
       set -x
     fi
-    bazel run //tools/code_format:check_format -- fix "${1}"
-    ./tools/spelling/check_spelling_pedantic.py fix "$1"
+    bazel run //tools/code_format:check_format -- fix "$@"
```

Comment:
can we check a comand-line switch here to allow previous behavior of:

    ./tools/code_format/check_format.py fix "$1"
    ./tools/spelling/check_spelling_pedantic.py fix "$1"

Reply:
not sure iiuc how that could work - the only way this function can be invoked is in this script and it doesnt do that

Reply:
I am suggesting we add a new command-line option to this script, and check the value of that option here prior to running bazel.

Reply:
Oh I see we'd have to call `./tools/code_format/check_format.py fix` in a loop here, rather than having the loop at the bottom around `format_one`. That seems fine.

Reply:
i dont think we want the python script to be invoked directly ever - we want it to use bazel

Reply:
Why not? it worked well yesterday. It's much much faster than running bazel. 

I know you  might not want it but I'm OK risking divergence for my local workflow to get speed. I also noticed when running the bazel version that it has to restart bazel cause of different options compared to when I build. And then it has to re-start bazel again when I build. So I am not a fan of having bazel in the path in my workflow.

I'm totally fine with having bazel in my path for CI, and if I decide I want to go slow and get more accurate results.

Reply:
> It's much much faster than running baze

i dont think so

i think the only delay with doing this heremtically/reproducibly is that on first invokation you have to build the build tools

this way will lead to it being much faster in the end - but i doubt it impacts perf much at all after first run

Reply:
> I know you might not want it but I'm OK risking divergence for my local workflow to get speed.

the performance of the underlying script is absolutely terrible we are talking about cherries atop a huge cake

Reply:
Let me explain the usage model where the flow I had built works really well for me. I'll use the example of a PR I have in flight (blocked by not having clang-tidy): https://github.com/envoyproxy/envoy/pull/29181

This file has just 2 files in it, one of them a C++ file. It's in a client that doesn't have the new bazel dependency, so it rocks:
```
time tools/local_fix_format.sh -verbose -main
Checking all files that have changed since the main branch.
+ ./tools/code_format/check_format.py fix .clang-tidy
+ ./tools/spelling/check_spelling_pedantic.py fix .clang-tidy
Checked 0 file(s) and 0 comment(s), found 0 error(s).
+ ./tools/code_format/check_format.py fix source/common/upstream/load_balancer_impl.cc
+ ./tools/spelling/check_spelling_pedantic.py fix source/common/upstream/load_balancer_impl.cc
Checked 1 file(s) and 268 comment(s), found 0 error(s).
0.373u 0.245s 0:00.46 132.6%	0+0k 0+64io 1pf+0w
```

By default his tool only runs on the open files in the git client. Since there were none I used "-main" which checks all the files that have changed relative to github 'main'.

Because this is so fast I run it all the time, before any commit.

I think I can see the argument that the underlying script has poor performance when it has to iterate over 1000s of files. But I never use it like that. So the performance is excellent.

Reply:
can i suggest landing this and iterating

i think what _this_ does is a pretty clear improvement - mho is that its totally unwanted to maintain non-bazel paths - it just creates the kind of problems you have just flagged offline

regardless - i think this improvement stands up by itself - but im not a user of this script so happy to close if its unwanted

Reply:
OK I can believe you that this is an improvement over the current state, but for my usage model the previous update to use bazel defeated the purpose of this script.

I'm fine landing this and I'll quickly follow with a fix that allows me to run the python directly if you don't provide it here. The other option is for me to orphan this script and just use a personal copy, but I think others found this methodology useful also when I shared it.